tungsten_ci_utils/nexus_cleanup/clean_up.groovy

```diff
@@ -0,0 +1,117 @@
+    listOfComponents.reverseEach{comp ->
+
+        log.info("Processing Component - ${comp.name()}, version: ${comp.version()}");
+        log.info("----------");
+        def fruitEnd = "null";
+        for(i = 0; i < whitelistCore.length; i++){
+            def fruitJuice = whitelistCore.collect{item -> item.contains(comp.version())}
+            
+            if(fruitJuice[i] == true){
+                fruitEnd = "true";
+                log.info("Fruit Juice! "+ fruitJuice[i] + " " + comp.version());
+                log.info("----------");
+                return fruitEnd;
+            }else{
+            
+                def splited = comp.version();
+                def spl = splited.split("-");
+
+                for(j = 0; j < spl.length; j++ ){            
+                    for(k = 0; k < whitelist.length; k++){
+                        def fruit = whitelist.collect{item -> item.contains(spl[j])}
+                        if(fruit[k] == true){
+                            fruitEnd = "true";
+                            log.info(spl[j]+ " true");
+                            return fruitEnd;
+                        }else{
+                            fruitEnd = "false";
+                            log.info(spl[j]+ " false");
+                        }
+                    }
+                }
+            }
+        }
+        log.info("----------");
+        log.info(fruitEnd + " Result");
+        log.info("----------");
+
+
+        if(fruitEnd == "false"){
+            log.info("am I?");
+            log.info(previousComp);
+
+            if(previousComp.equals(comp.name())) {
+            // for test purposes check of prev name commented out
+            
+            // if(fruitEnd == "false"){
+                compCount++;
+                log.info("ComCount: ${compCount}, ReteCount: ${retentionCount}");
+
+                if (compCount > retentionCount) {
+                    log.info("CompDate: ${comp.lastUpdated()} RetDate: ${retentionDate}");
+                    if(comp.lastUpdated().isBefore(retentionDate)) {
+                        log.info("compDate after retentionDate: ${comp.lastUpdated()} isAfter ${retentionDate}");
+                        log.info("deleting ${comp.group()}, ${comp.name()}, version: ${comp.version()}");
+
+                        // ------------------------------------------------
+                        // uncomment to delete components and their assets
+                        // service.deleteComponent(repo, comp);
+                        // ------------------------------------------------
+
+                        log.info("component deleted");
+                        log.info("----------");
+                        deletedComponentCount++;
+                    }
+                }
+            } else {
+                compCount = 1;
+                previousComp = comp.name();
+            }
+        }else{
+            log.info("Component skipped: ${comp.name()}");
```

Comment:
This line does not make too much sense as it only contains the image name, not the actual tag that was skipped.